### DISCUSSION TITLE
Unscale and inverse log smooth estimates for plotting

### DISCUSSION THREAD

**Discussion by tgohzy** [gavinsimpson/gratia#229] [Q&A]
Hi there, 

I am trying to plot GAM smooth estimates. I had log-transformed my predictor variable and also scaled it before running the model (there were a couple other variables in the model). 

I've used gratia::smooth_estimates to extract the model estimates and CIs. I think I managed to unscale the variable but I'm not quite sure if I managed to inverse log it. The plot (attached) definitely does not look right... These are the codes,

```
# Get a tibble of smooth estimates from the model mld7p
mld7p.est <- gratia::smooth_estimates(mld7p) %>%
  add_confint()

# Add a new column for the unscaled and inverse log  mld 
mld7p.est.unscaled <- mld7p.est %>% mutate(mld_7km_period = mean(HPSightingsMLD$mld_7km_period) + 
                                             (exp(mld_7km_period) * sd(HPSightingsMLD$mld_7km_period)))

# Plot
mld7p.plot <- mld7p.est.unscaled %>%
  add_confint() %>%
  filter(smooth == "s(mld_7km_period)") %>%
  ggplot() +
  geom_rug(aes(x = mld_7km_period),
           data = HPSightingsMLD,
           sides = "b", length = grid::unit(0.02, "npc")) +
  geom_ribbon(aes(ymin = lower_ci, ymax = upper_ci, x = mld_7km_period),
              alpha = 0.2) +
  geom_line(aes(x = mld_7km_period, y = est), lwd = 0.5) +
  labs(x = expression("MLD7p"~(m)), y = "Partial effect", title = "g") +
  theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))
```

HPSightingsMLD consist of the raw data (not log-transformed nor scaled). 

Hopefully someone can point out to me what I'm missing! Many thanks in advance :)

[mld.pdf](https://github.com/gavinsimpson/gratia/files/12304381/mld.pdf)

**Comment by tgohzy** [comment]
Hi Gavin,

Thanks so much for your reply. I really enjoyed the GAM video you made on YouTube, it has been very useful for me. 

I managed to get the plots I needed before you sent your reply. I needed to back-transform the data and had my ‘exp()’ at the wrong place in the R code. 

I tried the codes you suggested and plot it out but it looked really strange. Could you explain or direct me to a resource that explains the codes? I had used a binary model.

Many thanks,
Tiff

**Reply by gavinsimpson** [reply]
Oops; my bad. I misread the intent of your question when I was scanning things in the GitHub app on my phone. I thought you were back-transforming the y-axis to the response scale..
